From 552a7ed297653fb3ee66b94b9908a7e4e4dde7f4 Mon Sep 17 00:00:00 2001
From: Ramya Gnanasekar <quic_rgnanase@quicinc.com>
Date: Thu, 17 Nov 2022 10:43:41 +0530
Subject: [PATCH] ath12k: Fix double free during BE MU-MIMO htt stats

Removing the double free of buffer when parsing MU-MIMO
HTT Stats for 11be

Fixes: I89eab6c9 (ath12k: Add 11BE MU MIMO and OFDMA STATS support in
HTT STATS)

Signed-off-by: Ramya Gnanasekar <quic_rgnanase@quicinc.com>

Index: b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
===================================================================
--- a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c	2022-11-17 11:30:39.427031025 +0530
+++ b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c	2022-11-17 11:32:03.402377074 +0530
@@ -4484,9 +4484,6 @@ htt_print_be_ul_ofdma_trigger_stats(cons
 				 "be_ul_ofdma_rx_gi[%u] = %s ", j, rx_gi[j]);
 	}
 
-	for (i = 0; i < HTT_RX_PDEV_STATS_NUM_GI_COUNTERS; i++)
-		kfree(rx_gi[i]);
-
 	PRINT_ARRAY_TO_BUF(buf, len, htt_trigger_stats_buf->be_ul_ofdma_rx_nss,
 			   "be_ul_ofdma_rx_nss",
 			   HTT_RX_PDEV_STATS_NUM_SPATIAL_STREAMS, "\n\n");
@@ -4631,9 +4628,6 @@ htt_print_tx_pdev_rate_stats_be_ofdma_tl
 				 "be_ofdma_tx_gi[%u] = %s ", j, tx_gi[j]);
 	}
 
-	for (i = 0; i < HTT_TX_PEER_STATS_NUM_GI_COUNTERS; i++)
-		kfree(tx_gi[i]);
-
 	stats_req->buf_len = len;
 fail:
 	for (i = 0; i < HTT_TX_PEER_STATS_NUM_GI_COUNTERS; i++)
